### PR TITLE
Fix to Issue 13: Provide exhaustive stop word lists for all the supported languages

### PR DESCRIPTION
**Pull request by Warun26**
Hi,
I have tried to address Issue #13  in the fact-extractor which is to create a list of stop words for languages supported by tree tagger. I compiled the list of stop words for various languages using RanksNL[1] and stop words[2]. 

The following changes were made/added:
- Adds stop words for languages Bulgarian, Chinese, Dutch, English, French, German, Italian, Polish, Portugese, Russian, Slovak, Spanish. I have used the existing stop word list for Italian but made 4 new additions(invece, sembrava, senza, solo, stati).

- Adds a StopWords class to provide an API for accessing stop words for a specific language.

The stop words can be accessed in the following manner(The NLTK syntax for accessing stop words).
from stopwords import StopWords
italian_stop_words = StopWords.words('italian')

Please provide feedback as this is my first pull request ever!

Thanks,
Varun

Sources:
[1] -> http://www.ranks.nl/stopwords
[2] -> https://code.google.com/p/stop-words/

### CODE REVIEW

File: stopwords.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    #directory containing the above files
+    _stopword_directory = os.path.join('resources', 'stop-words')
+
+    @classmethod
+    def words(self,language, additional_stop_list = []):
+        #Check if we have a list of stop words for the requested language. If yes, read the corresponding file
+        if language in self._language_to_file_mapping:
+            stopWordFile = self._language_to_file_mapping[language]
+            pathToStopWordFile = os.path.join(self._stopword_directory, stopWordFile)
+            stopWordList = [word.strip() for word in codecs.open(pathToStopWordFile, "r", "utf-8")]
```
Comment (marfox) [inline stopwords.py]: I see you are stripping the stopwords, :+1: . While reading through the stopword lists, I was about to tell you that!

**Comment by marfox** [comment]
Good job @Warun26!
However, I'm not sure about the reliability of the resources you used.
Can you check if NLTK already provides stopword lists?
Thanks, keep up with the good work.

**Comment by aaiijmrtt** [comment]
hi @Warun26 ,
would you like some help with this issue? I was working on it yesterday, and I came across the same sources. a few languages were missing. further search lead to:

(i) Estonian:
https://github.com/tpetmanson/estnltk/blob/master/estnltk/textclassifier/analyzer.py
https://github.com/DGaffney/140kit_sandbox/blob/master/code/analyzer/resources/language/stop-words/estonian.txt
http://www.filewatcher.com/m/stopwords.txt.89-0.html

(ii) Latin:
https://github.com/6/stopwords/blob/master/dist/la.json

(iii) Mongolian:
https://github.com/nyamba/stop-words-mn/blob/master/stop-words-mn.txt

you might find them useful, but I cannot vouch for their authenticity. Swahili still remains to be salvaged.

**Comment by Warun26** [comment]
Hi @marfox,
Thanks for the encouraging feedback. I agree with you on the reliability of the resources. 
I did check out NLTK and they provide a corpus of stop words to most of the languages required by the TreeTagger. The exceptions being Bulgarian, Chinese, Estonian, Latin, Slovak, and Swahili. I also noticed that the Italian stop words used in bag_of_words.py is similar to the one used in NLTK. I believe NLTK to be a more reliable resource and I merely used the resources above as they were updated recently. Here's what I think is a better solution: 
1. Use the stop words provided by NLTK for Dutch, English, French, German, Italian, Portuguese, Russian and Spanish.
2. Use the stop words from the above resources for Bulgarian, Chinese, Polish, and Slovak
What are your thoughts?

**Comment by Warun26** [comment]
Hi @aaiijmrtt,
Thanks for helping me out. I will look into the above resources for Latin and Estonian.

**Comment by marfox** [comment]
That makes sense, feel free to proceed. 
WRT the languages not supported by NLTK, maybe a deeper search + merging different resources will raise the overall quality.

----- Reply message -----
Da: "Varun Varadarajan" <notifications@github.com>
A: "dbpedia/fact-extractor" <fact-extractor@noreply.github.com>
Cc: "Marco Fossati" <fossati@spaziodati.eu>
Oggetto: [fact-extractor] Fix to Issue 13: Provide exhaustive stop word lists for all the supported languages (#21)
Data: mer, mar 18, 2015 20:26
Hi @marfox,

Thanks for the encouraging feedback. I agree with you on the reliability of the resources. 

I did check out NLTK and they provide a corpus of stop words to most of the languages required by the TreeTagger. The exceptions being Bulgarian, Chinese, Estonian, Latin, Slovak, and Swahili. I also noticed that the Italian stop words used in bag_of_words.py is similar to the one used in NLTK. I believe NLTK to be a more reliable resource and I merely used the resources above as they were updated recently. Here's what I think is a better solution: 

1. Use the stop words provided by NLTK for Dutch, English, French, German, Italian, Portuguese, Russian and Spanish.

2. Use the stop words from the above resources for Bulgarian, Chinese, Polish, and Slovak

What are your thoughts? 

&mdash;

**Pull request closed without merging**